status: force linkage via call to je_zone_register

tbg opened the pull request:
go 1.9.4 and up require the env variable CGO_LDFLAGS_ALLOW to be set to
allow passing most linker flags. Instead of putting up with this, use
the alternative method of calling the hook directly to force inclusion.

Release note (build change): CockroachDB now builds with go 1.9.4 and
higher.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/22567)
<!-- Reviewable:end -->

petermattis commented:
nit: this is required for go1.9.4 as well.

@benesch will have to take a look at the change as I've lost/forgotten my make knowledge.

---

Review status: 0 of 2 files reviewed at latest revision, all discussions resolved, some commit checks pending.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/22567#-:-L50Xo4UnJiP3ah1IvUU:b-uoiy9q)*
<!-- Sent from Reviewable.io -->

bdarnell commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

This will unfortunately require this environment variable to be set any time you do a build outside our makefile (for editors, etc). 

Is `ld -u` the only way we can reference this symbol? We might be able to do something to reference the symbol from a bit of C code (and make sure somehow that it doesn't get optimized away). 

---

Review status: 0 of 2 files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/22567#-:-L524rXfYQx8OC-buRxG:b507c5y)*
<!-- Sent from Reviewable.io -->

tbg committed:
status: force linkage via call to je_zone_register

go 1.9.4 and up require the env variable CGO_LDFLAGS_ALLOW to be set to
allow passing most linker flags. Instead of putting up with this, use
the alternative method of calling the hook directly to force inclusion.

Release note (build change): CockroachDB now builds with go 1.9.4 and
higher.

tschottdorf commented:
@benesch great idea, PTAL.

bdarnell commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

We'll want to cherry-pick this into release-1.1 (and maybe release-1.0, although any future releases from that branch will presumably just use the builder image and we don't need to upgrade that to 1.8.7)

---

Review status: 0 of 2 files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/22567#-:-L59kKZM5TONDpM8zBp9:b-u14i3v)*
<!-- Sent from Reviewable.io -->

tschottdorf commented:
Not at all, thanks!

On Mon, Feb 12, 2018, 15:09 Nikhil Benesch <notifications@github.com> wrote:

> Closed #22567 <https://github.com/cockroachdb/cockroach/pull/22567>.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/cockroachdb/cockroach/pull/22567#event-1470268257>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AE135ErHwH4GpML0R5dMMn4QoInFOpivks5tUJp7gaJpZM4SBFMh>
> .
>


-- 

-- Tobias
